### ISSUE TITLE
Issue when changing baseItemType of a group from Number to String

### ISSUE THREAD

**Issue by goligo** [openhab/openhab-core#3266]
OpenHAB version 3.4.0, running on OpenHabian on a Raspberry 3.

I was changing multiple groups from baseItemType "Number" to "String". This was necessary as in the Shelly binding the profile type was changed. The possible values are 0, 1 and 2.

After doing this, all of these groups showed their value as DateTime, like "1970-01-01T01:00+01:00". No matter what I put in the state description into pattern or options, it was formatted like this, in the item when watched in the admin tool, as well as in the basic UI.

The reason was that in the functionName was not properly changed to "EQUALITY", when switching the baseItemType to "String", and it did stay on "MIN". This seems to confuse the framework and makes it interpret the result as a timestamp.

```
  "gHaupthaus_WC_Profil": {
    "class": "org.openhab.core.items.ManagedItemProvider$PersistedItem",
    "value": {
      "baseItemType": "String",
      "groupNames": [
        "gHaupthaus_Profil"
      ],
      "itemType": "Group",
      "tags": [],
      "label": "WC Profil",
      "category": "group",
      "functionName": "MIN"
    }
  },
```

As a workaround, I had to switch to baseItemType "None" first, and then to "String", then the functionName is also changed and everything works as expected.

**Comment by goligo** [comment]
As far I understand the code, this is the relevant code section - when changing the base item type, it tries to preserve the group function, regardless of whether the group function makes sense or can be applied to the new base type.

https://github.com/openhab/openhab-core/blob/3.4.x/bundles/org.openhab.core/src/main/java/org/openhab/core/items/ManagedItemProvider.java#L227-L230

The UI does not allow setting a group function for base item type String, so it should always be set to EQUALITY, when base item type is changed to String. I wonder if a generic solution is possible, where each type contains a default group function and a list of supported group functions, and in case the given group function is not contained in the list, it is set to the default group function for the type instead.

**Comment by J-N-K** [comment]
There is no validation at all if the configured group function is applicable to the item type. You can configure

```
Group:String:AVG gString "String Group"
```

in an `.items` file and no warning or similar is logged.

**Comment by goligo** [comment]
Thanks for looking into this. In the items file I can see there is a wrong function and I can change it. Both is not the case when using the Config UI:

Creating the Group with base type Number:
![Screenshot 2022-12-27 at 20 48 05](https://user-images.githubusercontent.com/1894593/209715850-121207fb-eb27-4cfc-8452-8263cc0bf0bc.png)

Changing the Group to base type String - aggregation function is hidden, so I can neither see that it is wrong, nor can I change it:
![Screenshot 2022-12-27 at 20 47 34](https://user-images.githubusercontent.com/1894593/209715989-e15a0831-6f66-4e38-a3de-371057376474.png)

But this is the resulting metadata, causing the symptoms I described above - a DecimalType is returned which is then converted into a DateTimeType ;-)
![Screenshot 2022-12-27 at 18 35 22](https://user-images.githubusercontent.com/1894593/209716076-8cc84dff-170a-431e-b1d0-3f928b797aff.png)

**Comment by J-N-K** [comment]
Yes, but a wrong group function should not be accepted at all, no matter how that was configured. Even if MainUI does not allow it, you can set a wrong type via REST API.

**Comment by goligo** [comment]
Absolutely, should be fixed in the underlying model, not the UI. I just wanted to point out, why it was especially painful to understand the issue, when using MainUI configuration instead of the items file.

**Comment by lionhe1966** [comment]
I am experiencing a related problem, linked to the fact that the UI does not allow any aggregation function (such as COUNT) when the basetype is string, while it is possible to specify it in .items files.
I explained the problem in the forum at this [link](https://community.openhab.org/t/oh3-user-interface-equivalent-of-groupcount-string/144133?u=lionello_marrelli).
Briefly, I have not found a way to insert through the UI a group of string items, as defined in an Item file
`Group:Number:COUNT("ControlledOff") gENEL_shred  "number of Shred equipments" `

Should I file this as a separate issue?
